### PR TITLE
functors_task_1 Chyruk

### PR DESCRIPTION
**Pull request by ChNokia**

### CODE REVIEW

File: functors_task_1/student/include/impl.h [hunk 1]
```diff
@@ -8,6 +8,17 @@
  */
 struct BeerOrganizer
 {
+    BeerBrand operator()() {
+        if (m_currentIndex == m_maxIndex - 1) {
+            m_currentIndex = static_cast<size_t>(BeerBrand::None);
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h:13]: you can also extract "None" to "min" variable

File: functors_task_1/student/include/impl.h [hunk 2]
```diff
@@ -16,16 +27,18 @@ struct BeerOrganizer
  *
  * @note Only Corona and HoeGaarden are expensive
  */
-bool isExpensiveBeer(/**???*/)
+bool isExpensiveBeer(BeerBrand brand)
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h:30]: _mark everything as const by default_

File: functors_task_1/student/include/impl.h [hunk 3]
```diff
@@ -16,16 +27,18 @@ struct BeerOrganizer
...
 {
+    return brand == BeerBrand::Corona || brand == BeerBrand::HoeGaarden;
 }
 
 /**
  * @todo Implement lambda beer country equality comparator
  * @return true if beer county is the same, false otherwise
  */
-auto sameCountry = [](/**???*/)
+auto sameCountry = [](BeerBrand brand1, BeerBrand brand2)
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h:39]: _const_

File: functors_task_1/student/include/impl.h [hunk 4]
```diff
@@ -40,9 +53,19 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(AlcoholDrink alcohol, NonAlcoholDrink nonAlcohol)
     {
+        if (alcohol == AlcoholDrink::Gin && nonAlcohol == NonAlcoholDrink::LimeJuice) {
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h:58]: _prefer switch_

**Review by smay1613 — COMMENTED** [review]
Needs changes, but pretty nice.

File: functors_task_1/student/include/impl.h (hunk 2 above)
Comment (ChNokia) [reply functors_task_1/student/include/impl.h]: done

File: functors_task_1/student/include/impl.h (hunk 3 above)
Comment (ChNokia) [reply functors_task_1/student/include/impl.h]: done

**Pull request closed without merging**